src/store/store/actions.js

```diff
@@ -18,3 +17,55 @@ function storesFetched(stores) {
     stores
   };
 }
+
+export const addStore = (newStore, userId, productId) => dispatch => {
+  request
+    .post(`${baseUrl}/store`)
+    // .set("Authorization", `Bearer ${token}`)
+    .send(newStore)
+    .then(res => {
+      const storeId = res.body.id;
+      const connection = { userId, productId, storeId };
+
+      request
+        .post(`${baseUrl}/connect`)
+        .send(connection)
+        .catch(console.error);
+      dispatch(storeAdded(res.body));
```

Comment:
I haven't looked at the backend yet but this seems like something the backend could handle by itself without needing a second request from the front-end, I'll keep an eye out in the server for this routes to be sure

Reply:
You could on the first post send like: 
`axios.post(`${baseUrl}/store`).set("Authorization", `Bearer ${token}`).send({ newStore, userId, productId })`
And then create the Connection in the same route as you are creating the store. Minimising requests is always a easy way of gaining performance